src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -914,9 +926,20 @@ export class BundleDataClient {
                 bundleInvalidFillsV3.push(fill);
                 return;
               }
+              // If the fill's repayment address is not a valid EVM address and the repayment chain is an EVM chain, the fill is invalid.
+              if (chainIsEvm(fill.repaymentChainId) && !utils.isAddress(fill.relayer)) {
+                const fillTransaction = await originClient.spokePool.provider.getTransaction(fill.transactionHash);
+                const originRelayer = fillTransaction.from;
+                // Repayment chain is still an EVM chain, but the msg.sender is a bytes32 address, so the fill is invalid.
+                if (!utils.isAddress(originRelayer)) {
```

Comment:
How could the repayment chain be EVM but the sender is not a valid address? I understand including this here for completeness, though maybe it should be an assert instead of an if, since it seems impossible to happen in reality

Reply:
The check is on the origin chain. There could technically be a fill on a non-evm chain with repayment on an evm chain.

Reply:
yeah i guess this will never be possible while all chains are evm. This is a fine placeholder for now i guess